spec/ghp_metadata_generator_spec.rb

```diff
@@ -4,8 +4,61 @@
 
 RSpec.describe(Jekyll::GitHubMetadata::GHPMetadataGenerator) do
   subject { described_class.new }
+  let(:source) { File.expand_path("../test-site", __FILE__) }
+  let(:dest) { File.expand_path("../../tmp/test-site-build", __FILE__)}
+  let(:user_config) { {} }
+  let(:configuration) do
+    merged_config = user_config.merge({source: source, dest: dest, quiet: true})
+    Jekyll.configuration(merged_config)
+  end
+  let(:site) { Jekyll::Site.new(configuration) }
 
   it "is safe" do
     expect(described_class.safe).to be(true)
   end
+
+  context "generating" do
+    before { stub_api("/repos/jekyll/github-metadata" , "repo") }
+    before { stub_api("/repos/jekyll/github-metadata/pages" , "repo_pages") }
+    before { stub_api("/repos/jekyll/jekyll.github.io" , "repo") }
+    before { stub_api("/repos/jekyll/jekyll.github.io/pages" , "repo_pages") }
+    before { stub_api("/repos/jekyll/jekyll.github.com" , "repo") }
+    before { stub_api("/repos/jekyll/jekyll.github.com/pages" , "repo_pages") }
```

Comment:
I'd put all of these into one `before(:each)` block. Should be a bit faster. Same with the env vars below.